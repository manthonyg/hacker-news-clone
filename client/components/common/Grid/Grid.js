import styled from "styled-components";

const Grid = styled.div`
  display: grid;
  overflow: hidden;
  grid-template-columns: ${(props) => props.cols};
  grid-template-rows: ${(props) => props.rows};
`;

export default Grid;
