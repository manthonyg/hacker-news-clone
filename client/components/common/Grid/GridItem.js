import styled from "styled-components";

const GridItem = styled.div`
  display: grid;
  grid-column-start: ${(props) => props.col};
  grid-row-start: ${(props) => props.row};
`;

export default GridItem;
