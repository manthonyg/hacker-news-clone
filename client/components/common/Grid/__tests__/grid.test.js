import React from "react";
import renderer from "react-test-renderer";
import Grid from "../Grid";

it("renders correctly", () => {
  const tree = renderer
    .create(<Grid cols="95% 5% auto" rows="10% 80% auto"></Grid>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
