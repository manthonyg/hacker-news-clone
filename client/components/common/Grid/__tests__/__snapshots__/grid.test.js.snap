// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm lmzCKR"
  cols="95% 5% auto"
  rows="10% 80% auto"
/>
`;
